src/core-services/orders/queries/ordersByAccountId.js

```diff
@@ -16,12 +16,17 @@ export default async function ordersByAccountId(context, { accountId, orderStatu
   const { collections } = context;
   const { Orders } = collections;
 
-  if (!accountId) {
-    throw new ReactionError("invalid-param", "You must provide accountId arguments");
+  if (!accountId) throw new ReactionError("invalid-param", "You must provide accountId arguments");
+
+  // Validate user has permission to view orders for all shopIds
+  if (!shopIds) throw new ReactionError("invalid-param", "You must provide ShopId(s)");
```

Comment:
I moved this check higher up in the function.

In the past, we did not require shopIds, it was an optional field, but with the 3.0 changes, we have been requiring shopIds to be passed everywhere. Because of this, I can check this higher up in the function, and also remove what was line 36, and add it to line 29 here instead, because we will always have a shopId available.

Does that all make sense? Or should I still do the check that was line 36?

Reply:
I think its fine to do the `shopId` check early since its always required.